### PR TITLE
Processes for Downloads

### PR DESCRIPTION
**Pull request by dcelik**
This PR is quite full of changes ranging from bug fixes, to sweeping type changes, and extra processes to prevent GUI freezing. The details are below:
* Adds CallbackProcess class for creating a new process to run a given function at a given interval. A Timer thread is used to check the progress of the process until some condition is met. It is designed to be very general and expected to be subclassed to fit general API call types.
* Adds SingleAPICallbackProcess class that inherits from CallbackProcess and is designed to work with a single API call that can return data or an error.
* GUI calls to getTestConfig are now implemented via the SingleAPICallbackProcess class.
* Modifies our request calls to use JSON instead of form-data encoding. This makes more sense as our payload is a dictionary, which lends itself very well to JSON style encoding.
* A lot of print statements were removed that added unnecessary clutter to our terminal.
* All API methods now return a set type: A 3-tuple of form (success, error, data).
* Fixed long-standing bug in how parse error reads error messages
* Consolidated writing to a path to helper function to minimize code duplication
* Remove JSON dumps from homography, this is designed to address some of the changes made in [this PR](https://github.com/santosfamilyfoundation/SantosCloud/pull/95) on SantosCloud

I would highly recommend understanding how the CallbackProcess works in order to use it for multiprocessing in our code.

### CODE REVIEW

File: application/app.py [hunk 1]
```diff
@@ -205,27 +222,37 @@ def test_object_callback(self, error_message):
...
             if os.path.exists(video_path):
                 self.roadusers_tracking_video_player.openFile(video_path)
 
-    def get_object_video(self):
-        success, err = api.getTestConfig(get_identifier(), 'object', os.path.join(get_project_path(), 'object_video'))
-        if success:
-            self.open_object_video()
-        else:
-            self.show_error(err)
+######################################################################################################
```
Comment (segerphilip) [inline application/app.py:247]: What is this for?

Comment (segerphilip) [reply application/app.py:247]: Maybe include a comment for why this divider is used.

File: application/custom/videographicsitem.py [hunk 2]
```diff
@@ -164,7 +164,7 @@ def mediaStatusChanged(self, status):
 
     def positionChanged(self, position):
         self.positionSlider.setValue(position)
-        print self.positionSlider.value()
+        #print self.positionSlider.value()
```
Comment (segerphilip) [inline application/custom/videographicsitem.py:167]: Remove if no longer needed, or add verbosity tags for log level?

**Review by segerphilip — APPROVED** [review]
Looks good to me, 🚢  if tested

**Review by jabb1123 — APPROVED** [review]

**Pull request merged**